### PR TITLE
Make it so Helix runs can be rerun and preserve history

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Addresses https://github.com/aspnet/AspNetCore/issues/8235

### CODE REVIEW

**Review by dougbu — APPROVED** [review]

**Comment by natemcmaster** [comment]
So, it looks like helix.proj is evaluated multiple times, so the 'hh:mm' approach results in the tests getting split accross many runs.

image

File: eng/helix/helix.proj [hunk 1]
```diff
@@ -19,7 +19,7 @@
   <PropertyGroup Condition=" '$(CI)' == 'true' ">
     <HelixType>ci</HelixType>
     <Creator>aspnetcore</Creator>
-    <HelixBuild>$(BUILD_BUILDNUMBER)</HelixBuild>
+    <HelixBuild>$(BUILD_BUILDNUMBER)-$([System.DateTime]::Now.ToString('HH:mm'))</HelixBuild>
```
Comment (natemcmaster) [inline eng/helix/helix.proj]:
```suggestion
    <HelixBuild>$(BUILD_BUILDNUMBER).$([System.DateTime]::Now.ToString('HH.mm'))</HelixBuild>
```


Let's workaround a bug in the Helix SDK causing it to generate an incorrect url in the logs.


image
https://dev.azure.com/dnceng/public/_build/results?buildId=197549&view=logs&j=1074ac01-8c55-5109-1085-a16f28133b94&t=c1387476-520f-5f83-69e0-e1e27b3446ff&l=1411

Helix says: https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190522.28.07~3A01
Actual URL: https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190522.28.07:01

File: eng/helix/helix.proj [hunk 2]
```diff
@@ -19,7 +19,7 @@
   <PropertyGroup Condition=" '$(CI)' == 'true' ">
     <HelixType>ci</HelixType>
     <Creator>aspnetcore</Creator>
-    <HelixBuild>$(BUILD_BUILDNUMBER)-$([System.DateTime]::Now.ToString('HH:mm'))</HelixBuild>
+    <HelixBuild>$(BUILD_BUILDNUMBER).$([System.DateTime]::Now.ToString('HH.mm'))</HelixBuild>
```
Comment (dougbu) [inline eng/helix/helix.proj]: Will this help the multiple names we've seen?

**Comment by JunTaoLuo** [comment]
Oops, gonna revert this.

**Pull request merged**